run.py

```diff
@@ -415,7 +415,8 @@ def main(self):
                     logger.error(f"❌ Failed on {self.env.record['instance_id']}: {e}")
                 else:
                     logger.error("❌ Failed on unknown instance")
-                self.env.reset_container()
+                # self.env.reset_container()
```

Comment:
**[swe-agent bug]** It should not reset like that on error:

* In a single-instance case, we don't want the container to restart (and even fail to do so!).
* If there are multiple instances, the next call to `run()` calls `env.reset` anyway.
* TODO: There is more cleanup needed, since reset_container uniquely calls some `init` functions for very very strange reasons. Those should be moved to `env.setup`.